dsc/workflows/base/__init__.py

```diff
@@ -294,39 +313,56 @@ def process_sqs_queue(self) -> list[tuple[str, dict]]:
         sqs_client = SQSClient(
             region=CONFIG.aws_region_name, queue_name=self.output_queue
         )
+
         items = []
         for sqs_message in sqs_client.receive():
             try:
-                item_identifier, message_body = sqs_client.process_result_message(
+                item_identifier, result_message = sqs_client.process_result_message(
                     sqs_message
                 )
-                items.append((item_identifier, message_body))
-                logger.debug(item_identifier, message_body)
-                self.report_data.append(f"{item_identifier}: {message_body}")
             except Exception:
                 error_message = f"Error while processing SQS message: {sqs_message}"
                 logger.exception(error_message)
-                self.report_data.append(error_message)
+                self.workflow_events.errors.append(error_message)
                 continue
+
+            # capture all processed items, whether ingested or not
+            item_data = {
+                "item_identifier": item_identifier,
+                "result_message": result_message,
+                "ingested": result_message["ResultType"] == "success",
+            }
+            self.workflow_events.processed_items.append(item_data)
+            items.append(item_data)
+
+            if not item_data["ingested"]:
+                message = (
+                    f"Item '{item_identifier}' did not ingest successfully: {sqs_message}"
+                )
+                logger.info(message)
+                self.workflow_events.errors.append(message)
```

Comment:
After our discussions last week, I had to do a double/triple take here, but this feels right.

What I like about this flow:
- we _always_ put the processed item into the `WorkflowEvents.processed_items` list, whether ingested or not
- we save human-friendly message strings to `WorkflowEvents.errors`

Perhaps as reports grow in the future, we might want to lean even heavier into that and call it something like `WorkflowEvents.error_messages` or something, but maybe not required.

What we should be careful about -- and maybe this even suggests a docstring sometime -- is not treating `Workflow.WorkflowEvents` as data to feed into more work with the items.  My understanding is that it's 100% about reporting... they just so happen to be structured objects.